File: src/main/java/interfaces/Identificacao.java
package interfaces;

public interface Identificacao {
    String getCnpj();
    Integer getPeriodo();

    static Identificacao of(final String cnpj, final Integer periodo) {
        return new Identificacao () {
            public String getCnpj() {
                return cnpj;
            }
            public Integer getPeriodo() {
                return periodo;
            }
        };
    }
}

File: src/main/java/interfaces/convertido/Identificacao.kt
package interfaces.convertido

 interface Identificacao {
    val cnpj: String
    val periodo: Int?

    companion object {
        fun of(cnpj: String, periodo: Int?): Identificacao {
            return object : Identificacao {
                override val cnpj: String
                    get() = cnpj
                override val periodo: Int?
                    get() = periodo
            }
        }
    }
 }




